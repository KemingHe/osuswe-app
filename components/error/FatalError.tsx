import { FireIcon } from '@heroicons/react/24/outline';
import type { JSX } from 'react';

export default function FatalError(): JSX.Element {
  return (
    <div
      className="card size-48"
      role="alert"
      aria-live="assertive"
      aria-atomic="true"
    >
      <figure className="pt-8">
        {/* Size matching the LoadingSpinner logo of 60px x 60px. */}
        <FireIcon className="w-[60px] h-[60px] text-error" aria-hidden="true" />
      </figure>
      <div className="card-body pt-4 text-center text-lg">
        <p className="h-full flex flex-col justify-center items-center leading-snug">
          <span className="sr-only">Service Unavailable</span>
          <span aria-hidden="true">Service</span>
          <span aria-hidden="true">Unavailable</span>
        </p>
      </div>
    </div>
  );
}
