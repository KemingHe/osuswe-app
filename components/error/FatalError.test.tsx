import { type RenderResult, render, screen } from '@testing-library/react';

import FatalError from '@/components/error/FatalError';

describe('FatalError component', () => {
  it('renders an accessible and visible fatal error message', () => {
    render(<FatalError />);

    // Select the element by its "alert" role.
    const elementByRole = screen.getByRole('alert');
    expect(elementByRole).toBeVisible();

    // Select the element by its text content.
    const elementByText = screen.getByText('Service Unavailable');
    expect(elementByText).toBeVisible();
  });

  it('matches the snapshot', () => {
    const { asFragment }: RenderResult = render(<FatalError />);
    expect(asFragment()).toMatchSnapshot();
  });
});
