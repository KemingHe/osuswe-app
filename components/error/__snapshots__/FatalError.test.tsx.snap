// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FatalError component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    aria-atomic="true"
    aria-live="assertive"
    class="card size-48"
    role="alert"
  >
    <figure
      class="pt-8"
    >
      <svg
        aria-hidden="true"
        class="w-[60px] h-[60px] text-error"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15.362 5.214A8.252 8.252 0 0 1 12 21 8.25 8.25 0 0 1 6.038 7.047 8.287 8.287 0 0 0 9 9.601a8.983 8.983 0 0 1 3.361-6.867 8.21 8.21 0 0 0 3 2.48Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
        <path
          d="M12 18a3.75 3.75 0 0 0 .495-7.468 5.99 5.99 0 0 0-1.925 3.547 5.975 5.975 0 0 1-2.133-1.001A3.75 3.75 0 0 0 12 18Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
    </figure>
    <div
      class="card-body pt-4 text-center text-lg"
    >
      <p
        class="h-full flex flex-col justify-center items-center leading-snug"
      >
        <span
          class="sr-only"
        >
          Service Unavailable
        </span>
        <span
          aria-hidden="true"
        >
          Service
        </span>
        <span
          aria-hidden="true"
        >
          Unavailable
        </span>
      </p>
    </div>
  </div>
</DocumentFragment>
`;
